### ISSUE TITLE
Ensure panic docs are correct

### ISSUE THREAD

**Issue by jendrikw** [rust-lang/rust-clippy#9333]
### What it does

Functions with a heading `# Panics` in their documentation [as described in RFC 1574](https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md) should have at least one doctest with `should_panic` or at least one `#[test] #[should_panic]` referencing it.

This lint would trigger if that is not the case.

### Lint Name

panic_doc_untested

### Category

pedantic, restriction

### Advantage

- Ensure code is tested and works as described in the documentation.

### Drawbacks

Might be noisy if the panic conditions are nontrivial.

### Example

```rust
/// Retrieves a copy of the `i`-th element of `slice`.
/// # Panics
/// Panics if `i` is outside bounds.
fn get(slice: &[u8], i: usize) -> u8 {
    slice[i]
}
```

Could be written as:

```rust
/// Retrieves a copy of the `i`-th element of `slice`.
/// # Panics
/// Panics if `i` is outside bounds.
/// ```should_panic
/// get(&['a', 'b', 'c'], 3);
/// ```
fn get(slice: &[u8], i: usize) -> u8 {
    slice[i]
}
```

**Comment by llogiq** [comment]
Shouldn't that be

```
/// Retrieves a copy of the `i`-th element of `slice`.
/// # Panics
/// Panics if `i` is outside bounds.
///
/// # Examples
/// ```should_panic
/// get(&['a', 'b', 'c'], 3);
/// ```
fn get(slice: &[u8], i: usize) -> u8 {
    slice[i]
}
```

**Comment by jendrikw** [comment]
Are you talking about the heading `# Examples`? Yes, that is recommended as per [RFC 505](https://github.com/rust-lang/rfcs/blob/ff29887f5ef12faaf0a0a60777a0c30aa5ee2014/text/0505-api-comment-conventions.md#using-markdown), but I tend to leave it off as all my doctests are examples.

**Comment by llogiq** [comment]
Yes. Our docs should follow the recommended style.

**Comment by jendrikw** [comment]
Ok, I updated the code.
